Update the plugin to the new logtash-core-plugin-api

ph opened the pull request:
Ref: https://github.com/elastic/logstash/issues/5269

ph committed:
Update the plugin to the new logtash-core-plugin-api

Ref: https://github.com/elastic/logstash/issues/5269

purbon commented on logstash-input-example.gemspec:
```diff
@@ -1,6 +1,6 @@
 Gem::Specification.new do |s|
   s.name = 'logstash-input-example'
-  s.version         = '2.0.4'
+  s.version = '1.0.0'
```
why downgrade version?

ph replied:
I thought this plugin wasn't publishing on rubygems but this isnt the case.

ph replied:
I will revert that.

ph committed:
revert version change

ph commented:
Update from review.

purbon commented:
LGTM

ph committed:
Update plugin with the latest logstash-core-plugin-api constraints

it should now use a range instead of a persimitic version on the patch
level.

```
  s.add_runtime_dependency "logstash-core-plugin-api", ">= 1.60", "<= 2.99"
```

ph commented:
@purbon I took another look at the dependency of this PR and sadly they were wrong we are now using a range on the `logstash-core-plugin-api` gem, I've updated this PR with the appropriate version.

ph commented:
I believe the version of this gem should be 1.0.0, I mean as a plugin author I would like to start with this version and not have to replace it.

I think the version changes shouldn't matter in this plugin since we shouldn't publish them.

ph commented:
Also, I've pushed a new version of the constraint for this plugin and fixed the tests. 
The plugin was missing a runtime dependency on the `logstash-codec-plain`, Its been like this for some time.

purbon commented:
```
LoadError: JRuby ext built for wrong Java version in `JrubyEventExtService': java.lang.UnsupportedClassVersionError: JrubyEventExtService : Unsupported major.minor version 52.0
  require at org/jruby/RubyKernel.java:1040
   (root) at /home/travis/build/logstash-plugins/logstash-input-example/vendor/bundle/jruby/1.9/gems/logstash-core-event-java-5.0.0.alpha5.snapshot1-java/lib/logstash-core-event-java/logstash-core-event-java.rb:25
  require at org/jruby/RubyKernel.java:1040
   (root) at /home/travis/build/logstash-plugins/logstash-input-example/vendor/bundle/jruby/1.9/gems/logstash-core-event-java-5.0.0.alpha5.snapshot1-java/lib/logstash-core-event.rb:1
```

wrong java version used for travis tests?

ph committed:
JDK8

ph commented:
@purbon travis is now green.
